packages/product-json-to-csv/test/map-headers.spec.js

```diff
@@ -0,0 +1,181 @@
+      },
+      {
+        label: '_hasStagedChanges',
+        value: 'hasStagedChanges',
+      },
+      {
+        label: 'productType',
+        value: 'productType',
+      },
+      {
+        label: 'variantId',
+        value: 'variant.id',
+      },
+      {
+        label: 'variantKey',
+        value: 'variant.key',
+      },
+      {
+        label: 'id',
+        value: 'id',
+      },
+      {
+        label: 'key',
+        value: 'key',
+      },
+      {
+        label: 'sku',
+        value: 'variant.sku',
+      },
+      {
+        label: 'tax',
+        value: 'taxCategory',
+      },
+      {
+        label: 'categories',
+        value: 'categories',
+      },
+      {
+        label: 'images',
+        value: 'variant.images',
+      },
+      {
+        label: 'name.en',
+        value: 'name.en',
+      },
+      {
+        label: 'description.en',
+        value: 'description.en',
+      },
+      {
+        label: 'slug.en',
+        value: 'slug.en',
+      },
+      {
+        label: 'attribute',
+        value: 'attribute',
+      },
+      {
+        label: 'customAttribute',
+        value: 'customAttribute',
+      },
+    ]
+
+    expect(mapHeaders(headers)).toEqual(expected)
+  })
+
+  it('maps own headers from own attributes', () => {
+    const headers = [
+      'publish',
+      'hasStagedChanges',
+      'productType',
+      'variant.id',
+      'variant.key',
+      'id',
+      'key',
+      'variant.sku',
+      'taxCategory',
+      'categories',
+      'variant.images',
+      'name.en',
+      'description.en',
+      'slug.en',
+      'attribute',
+      'customAttribute',
+    ]
+    const expected = [
+      {
+        label: '_published',
+        value: 'publish',
+      },
+      {
+        label: '_hasStagedChanges',
+        value: 'hasStagedChanges',
+      },
+      {
+        label: 'productType',
+        value: 'productType',
+      },
+      {
+        label: 'variantId',
+        value: 'variant.id',
+      },
+      {
+        label: 'variantKey',
+        value: 'variant.key',
+      },
+      {
+        label: 'id',
+        value: 'id',
+      },
+      {
+        label: 'key',
+        value: 'key',
+      },
+      {
+        label: 'sku',
+        value: 'variant.sku',
+      },
+      {
+        label: 'tax',
+        value: 'taxCategory',
+      },
+      {
+        label: 'categories',
+        value: 'categories',
+      },
+      {
+        label: 'images',
+        value: 'variant.images',
+      },
+      {
+        label: 'name.en',
+        value: 'name.en',
+      },
+      {
+        label: 'description.en',
+        value: 'description.en',
+      },
+      {
+        label: 'slug.en',
+        value: 'slug.en',
+      },
+      {
+        label: 'attribute',
+        value: 'attribute',
+      },
+      {
+        label: 'customAttribute',
+        value: 'customAttribute',
+      },
+    ]
+
+    expect(mapHeaders(headers)).toEqual(expected)
```

Comment:
These should likely really be split up. Such masive asserts are hard to debug, maintain and extend. Tricks up our sleeve are expect.arrayContaining and that in multiple its.

Reply:
I agree with this. Why I didn't split it up is that in this case the order of the array is also important